### ISSUE TITLE
doesnt work

### ISSUE THREAD

**Issue by MaherTayfour** [naztronaut/FCW#2]
sir please the entire file /var/www/html/fcw isnt avaliable so the enitre flask app (https://raspip/fcw) deosnt work i followed each step and still have nothing can you please explain ?

**Comment by naztronaut** [comment]
Did you set up ssl for your https connection?

What error are you seeing? Some screenshots would be helpful. And maybe some logs from `/var/log/apache2` - either access logs or error logs will be super helpful.

**Comment by MaherTayfour** [comment]
it just says ah00498 and ah00094 and ah00491 are you sure the fcw.conf file is correct ? ive seen it have fcw.wsgi but infact the only file with .wsgi is called led.wsgi so i modified the name i installed the folder and everything in var/www/html so now it is var/www/html/fcw  can u please check teh fcw.conf file writen in the repo ? becuase when i follow instructions even the default page of apache stops working. 
thank you

**Comment by naztronaut** [comment]
You're right that the file name is incorrect. You can either Rename `led.wsgi` to `fcw.wsgi` or in your configuration file, change this line:

```
WSGIScriptAlias /fcw /var/www/html/fcw/fcw.wsgi
```

to:

```
WSGIScriptAlias /fcw /var/www/html/fcw/led.wsgi
```

I just made that change in the repo, feel free to repull as well. Hope that helps! Also if you get more errors, post some of the error_log data here and I can take a look.

**Comment by MaherTayfour** [comment]
before i do 
a2ensite fcw.conf 
the default page works then after i execute that command the default page stops working and 
raspberryip/fcw doesnt work any idea why ? 
the log files of apache show only 3 error codes 
ah00489
ah00094
ah00491 
are you sure that other names / file path arent incorrect ? for example where should the index.html page be ? 
thank you

**Comment by naztronaut** [comment]
I've noticed that sometimes, Apache on the Pi can't handle multiple .conf files for some reason. I need to look deeper into why that's the case but for now, you should disable the default configuration and just run fcw.conf. There's a discussion on that here: https://www.raspberrypi.org/forums/viewtopic.php?t=260853

Not sure why this is the case. So for now, try disabling the default config:

```
sudo a2dissite 000-default.conf
```

And then enable fcw with the command you wrote above. The default page won't work but your fcw app should. 

All of your files should be in whatever is defined in the configuration, which is `/var/www/html/fcw/`

**Comment by MaherTayfour** [comment]
Hello sir i have another question i made a simple php script and replaced it with the apache default index.html 
In my script i have a simple :
exec(sudo python3 /home/pi/Rpi_ws281x/python/examples/strandtest.py) 
And its set to a button 
This oagw works and does show a button problem is the commad doesn't actually work you know whats the reason? Like it doesn't run the script i specified 
If you can help me it would be great.
Thank you.

**Comment by naztronaut** [comment]
It's probably because Apache doesn't have access to run the script as sudo. If you look at the error logs, you might see an error, it should be at:

```
/var/log/apache2/error_log
```

You can also look at `access_log` in case anything is logged there. 

I suspect that's the issue. I tried doing this a few years ago and I had to play around with permissions a lot to get sudo to work. I don't have that set up anymore so I forget what steps I took to fix it. Here's a good stackoverflow thread that may be helpful: 

https://stackoverflow.com/questions/60376585/how-to-run-php-exec-with-sudo-on-live-server

It's definitely not security-friendly. That's why it's hard to do. It's also why I did this project as a python script running in the background as a service with sudo before I touch it with Apache.
